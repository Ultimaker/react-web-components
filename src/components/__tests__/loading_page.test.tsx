// Copyright (c) 2018 Ultimaker B.V.
import * as React from 'react';
import { shallow } from 'enzyme';

// component
import LoadingPage from '../loading_page';
import Spinner from '../spinner';

describe('The LoadingPage component', () => {
    let wrapper;

    beforeEach(() => {
        jest.useFakeTimers();
        wrapper = shallow(<LoadingPage />);
    });

    afterEach(() => {
        jest.useRealTimers();
    });

    it('should spinner after timeout', () => {
        expect(wrapper.find(Spinner)).toHaveLength(0);
        jest.runOnlyPendingTimers();
        expect(wrapper.find(Spinner)).toHaveLength(1);
    });

    it('should cancel timeout on unmount', () => {
        wrapper.unmount();
        expect(clearTimeout).toBeCalled();
    });
});
