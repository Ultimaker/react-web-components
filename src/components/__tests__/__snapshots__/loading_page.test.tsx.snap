// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The LoadingPage component should render 1`] = `
<div
  className="loading-page"
>
  <Spinner />
</div>
`;
